src/setfit/data.py

```diff
@@ -15,69 +15,115 @@
-    if dataset == "emotion":
-        return {
-            "text": ["The sentence is sadness"] * sample_size
-            + ["The sentence is joy"] * sample_size
-            + ["The sentence is love"] * sample_size
-            + ["The sentence is anger"] * sample_size
-            + ["The sentence is fear"] * sample_size
-            + ["The sentence is surprise"] * sample_size,
-            "label": [0] * sample_size
-            + [1] * sample_size
-            + [2] * sample_size
-            + [3] * sample_size
-            + [4] * sample_size
-            + [5] * sample_size,
-        }
-    elif dataset == "ag_news":
-        return {
-            "text": ["The sentence is world"] * sample_size
-            + ["The sentence is sports"] * sample_size
-            + ["The sentence is business"] * sample_size
-            + ["The sentence is tech"] * sample_size,
-            "label": [0] * sample_size + [1] * sample_size + [2] * sample_size + [3] * sample_size,
-        }
-    elif dataset == "amazon_counterfactual_en":
-        return {
-            "text": ["The sentence is not counterfactual"] * sample_size
-            + ["The sentence is counterfactual"] * sample_size,
-            "label": [0] * sample_size + [1] * sample_size,
-        }
-    elif dataset == "SentEval-CR":
-        return {
-            "text": ["The sentence is negative"] * sample_size + ["The sentence is positive"] * sample_size,
-            "label": [0] * sample_size + [1] * sample_size,
-        }
-    elif dataset == "sst5":
-        return {
-            "text": ["The sentence is very negative"] * sample_size
-            + ["The sentence is negative"] * sample_size
-            + ["The sentence is neutral"] * sample_size
-            + ["The sentence is positive"] * sample_size
-            + ["The sentence is very positive"] * sample_size,
-            "label": [0] * sample_size + [1] * sample_size + [2] * sample_size + [3] * sample_size + [4] * sample_size,
-        }
-    elif dataset == "enron_spam":
-        return {
-            "text": ["The sentence is ham"] * sample_size + ["The sentence is spam"] * sample_size,
-            "label": [0] * sample_size + [1] * sample_size,
-        }
-    elif dataset == "tweet_eval_stance_abortion":
-        return {
-            "text": ["The sentence is none"] * sample_size
-            + ["The sentence is against"] * sample_size
-            + ["The sentence is favor"] * sample_size,
-            "label": [0] * sample_size + [1] * sample_size + [2] * sample_size,
-        }
-    elif dataset == "ade_corpus_v2_classification":
-        return {
-            "text": ["The sentence is not related"] * sample_size + ["The sentence is related"] * sample_size,
-            "label": [0] * sample_size + [1] * sample_size,
+def get_templated_dataset(
+    dataset: Dataset = None,
+    candidate_labels: List[str] = None,
+    reference_dataset: str = None,
```

Comment:
```suggestion
    dataset: Optional[Dataset] = None,
    candidate_labels: Optional[List[str]] = None,
    reference_dataset: Optional[str] = None,
```
Adding `Optional` in accordance to [PEP 484](https://peps.python.org/pep-0484/#union-types), I quote:
> A past version of this PEP allowed type checkers to assume an optional type when the default value is None, as in this code:
> ```
> def handle_employee(e: Employee = None): ...
> ```
> This would have been treated as equivalent to:
> ```
> def handle_employee(e: Optional[Employee] = None) -> None: ...
> ```
> This is no longer the recommended behavior. Type checkers should move towards requiring the optional type to be made explicit.